src/core/Table/TableRowMemoized.tsx

```diff
@@ -146,6 +146,7 @@ export const TableRowMemoized = React.memo(
     prevProp.rowProps === nextProp.rowProps &&
     prevProp.expanderCell === nextProp.expanderCell &&
     prevProp.tableHasSubRows === nextProp.tableHasSubRows &&
+    prevProp.state.columnOrder === nextProp.state.columnOrder &&
```

Comment:
I am afraid that there can be new `columnOrder` reference even it is the same. Maybe better check like:
```tsx
prevProp.state.columnOrder.every((column, index) => column === nextProp.state.columnOrder[index])
```

Reply:
@bentleyvk Columns' data are not being reordered after this change.